packages/bpk-component-slider/src/BpkSlider.js

```diff
@@ -0,0 +1,121 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import Slider from 'react-slider';
+import { cssModules } from 'bpk-react-utils';
+import React, { Component } from 'react';
+
+import STYLES from './bpk-slider.scss';
+
+const getClassName = cssModules(STYLES);
+
+class BpkSlider extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { value: props.value };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(value) {
+    this.setState({ value });
+    this.props.onChange(value);
+  }
+
+  render() {
+    const {
+      min,
+      max,
+      step,
+      large,
+      valueTemplate,
+      minDistance,
+      className,
+      onAfterChange,
+    } = this.props;
+
+    const sliderClass = [];
+    const handleClass = [];
+    const barClass = [];
+
+    sliderClass.push(getClassName('bpk-slider'));
+    sliderClass.push(getClassName(this.state.value.length
+      ? 'bpk-slider__range'
+      : 'bpk-slider__simple'));
+    handleClass.push(getClassName('bpk-slider-handle'));
+    barClass.push(getClassName('bpk-slider-bar'));
+
+    if (className) sliderClass.push(getClassName(className));
+    if (large) handleClass.push(getClassName('bpk-slider-handle--large'));
+
+    const handleActiveClass = getClassName('bpk-slider-handle--active');
+    return (
+      <div className={getClassName('bpk-slider-wrapper')}>
+        {this.state.value.length
+          ? valueTemplate(this.state.value[0], this.state.value[1])
+          : valueTemplate(min, this.state.value)}
+        <Slider
+          max={max}
+          min={min}
+          step={step}
+          value={this.state.value}
+          minDistance={minDistance}
+          withBars
+          className={sliderClass.join(' ')}
+          handleClassName={handleClass.join(' ')}
+          handleActiveClassName={handleActiveClass}
+          barClassName={barClass.join(' ')}
+          onChange={value => this.handleChange(value)}
+          onAfterChange={value => onAfterChange(value)}
+        />
+      </div>
+    );
+  }
+}
+
+BpkSlider.propTypes = {
+  min: PropTypes.number.isRequired,
+  max: PropTypes.number.isRequired,
+  minDistance: PropTypes.number,
+  value: PropTypes.oneOfType([PropTypes.array, PropTypes.number]).isRequired,
+  step: PropTypes.number,
+  className: PropTypes.string,
+  large: PropTypes.bool,
+  onChange: PropTypes.func,
+  valueTemplate: PropTypes.func,
+  onAfterChange: PropTypes.func,
+};
+
+BpkSlider.defaultProps = {
+  step: 1,
+  minDistance: 1,
+  className: '',
+  large: false,
+  valueTemplate: (min, max) => (
```

Comment:
We should avoid using the terminology `template` in react - everything should be a component. 

Can we rename this? Perhaps `ValueComponent` ?

Reply:
Also the signature should be updated to conform to an actual component i.e. `(props) => <Component />`

Reply:
In fact... I actually think that this "value template" area shouldn't be part of this component and should instead live in userland.

Apologies haha.

Reply:
Done